### PR TITLE
admin: Add socket options

### LINKED ISSUE #7524: admin: Using SO_REUSEPORT socket option
*Description*:
We plan to use `SO_REUSEPORT` to achieve graceful restart of Envoy (similar to #3804).
It's possible to set socket options for the application listener socket, however it looks like it's impossible to do so for the admin port.

Would it be acceptable to add configuration options to admin port for setting socket options?

### PR DESCRIPTION
**Pull request by euroelessar**
Description: Add ability to configure socket options for admin listener socket
Risk Level: LOW (opt-in change)
Testing: unit and integration test to ensure parsing and propagation of new protobuf field
Docs Changes: inline documentation
Release Notes: updated
Fixes #7524

### CODE REVIEW

**Comment by euroelessar** [comment]
@mattklein123 I need some guidance how to test this change:
It looks like admin interface is not abstracted enough to be able to test it without spawning the full envoy, but in that case it's impossible to mock/track syscalls and hard to observe potential side effects.

File: include/envoy/server/admin.h [hunk 1]
```diff
@@ -124,6 +124,7 @@ class Admin {
   virtual void startHttpListener(const std::string& access_log_path_,
                                  const std::string& address_out_path,
                                  Network::Address::InstanceConstSharedPtr address,
+                                 Network::Socket::OptionsSharedPtr socketOptions,
```
Comment (mattklein123) [inline include/envoy/server/admin.h]: nit: `const Network::Socket::OptionsSharedPtr& socket_options` here and elsewhere

File: docs/root/intro/version_history.rst [hunk 2]
```diff
@@ -3,6 +3,7 @@ Version history
 
 1.12.0 (pending)
 ================
+* admin: added ability to configure listener socket options
```
Comment (mattklein123) [inline docs/root/intro/version_history.rst]: please ref link to the new config field.

File: include/envoy/server/admin.h [hunk 3]
```diff
@@ -124,7 +124,7 @@ class Admin {
   virtual void startHttpListener(const std::string& access_log_path_,
                                  const std::string& address_out_path,
                                  Network::Address::InstanceConstSharedPtr address,
-                                 Network::Socket::OptionsSharedPtr socketOptions,
+                                 const Network::Socket::OptionsSharedPtr& socketOptions,
```
Comment (mattklein123) [inline include/envoy/server/admin.h]: nit: s/socketOptions/socket_options here and elsewhere.

**Comment by mattklein123** [comment]
Can you merge master? It should fix coverage.

/wait

**Comment by mattklein123** [comment]
Sorry I think CI is wedged. Can you push an empty commit or merge master again?

/wait

**Pull request merged**